src/main/java/com/conveyal/gtfs/loader/Table.java

```diff
@@ -478,13 +503,48 @@ public String generateSelectSql (String namespace, Requirement minimumRequiremen
         String fieldsString;
         String tableName = String.join(".", namespace, name);
         String fieldPrefix = tableName + ".";
-        if (minimumRequirement.equals(EDITOR)) fieldsString = commaSeparatedNames(editorFields(), fieldPrefix);
-        else if (minimumRequirement.equals(OPTIONAL)) fieldsString = commaSeparatedNames(specFields(), fieldPrefix);
-        else if (minimumRequirement.equals(REQUIRED)) fieldsString = commaSeparatedNames(requiredFields(), fieldPrefix);
-        else fieldsString = "*";
+        if (minimumRequirement.equals(EDITOR)) {
+            fieldsString = commaSeparatedNames(editorFields(), fieldPrefix, true);
+        } else if (minimumRequirement.equals(OPTIONAL)) {
+            fieldsString = commaSeparatedNames(specFields(), fieldPrefix, true);
+        } else if (minimumRequirement.equals(REQUIRED)) {
+            fieldsString = commaSeparatedNames(requiredFields(), fieldPrefix, true);
+        } else fieldsString = "*";
         return String.format("select %s from %s", fieldsString, tableName);
     }
 
+    /**
+     * Generate a select statement from the columns that actually exist in the database table
+     */
+    public String generateSelectAllExistingFieldsSql(Connection connection, String namespace) throws SQLException {
+        // select all columns from table
+        // FIXME This is postgres-specific and needs to be made generic for non-postgres databases.
+        PreparedStatement statement = connection.prepareStatement(
+            "SELECT column_name FROM information_schema.columns WHERE table_schema = ? AND table_name = ?"
```

Comment:
I have a concern about this method.  I'm not sure it's actually exporting whatever was loaded into the database from the GTFS feed.  It looks it's filtering the columns exported by fields it finds in the table definition. This means it would drop any extensions that are not a part of that definition.  This needs to be addressed.

We probably also need a test case to ensure that non-editor feeds do not lose extensions after a round trip (import -> export).

Reply:
I'm not sure I follow.  This was added to mimic existing behavior where all of the database fields were exported in the fashion of `select * from table`.  So the past way of doing things did basically the same thing except it didn't format certain fields.

Reply:
In this block of code, you're only including the column name if it is found in the canonical list of fields for the table.
https://github.com/conveyal/gtfs-lib/blob/05fe14e560ff81acdf567e02a0b988a7bff4978a/src/main/java/com/conveyal/gtfs/loader/Table.java#L538-L543

If I had a field called `date_color` in the `calendar_dates` table from a feed I imported, on export this field would be dropped.  This is currently only relevant for feeds that are never imported to the editor, where we don't currently guarantee that extensions will make the round trip.

Reply:
@abyrd I'm don't think this would work as you are suggesting.  Previously and currently, the Java models haven't been involved in the output to csv files.  The writing of the output happens here: https://github.com/conveyal/gtfs-lib/blob/05fe14e560ff81acdf567e02a0b988a7bff4978a/src/main/java/com/conveyal/gtfs/loader/JdbcGtfsExporter.java#L379-L397
As can be seen there is no middleman in the process, the csv is exported directly from postgres to an outputstream.

Reply:
Thanks @evansiroky I was not familiar with how this works so am just now understanding all the implications. Indeed if we want to preserve all the columns there's no real way to go through the Java model objects.